glitch/data.go

```diff
@@ -99,9 +99,9 @@ func (d *dataError) String() string {
 // FromHTTPProblem will create a DataError from an HTTPProblem
 func FromHTTPProblem(inner error, msg string) DataError {
 	switch ie := inner.(type) {
-	case *HTTPProblem:
+	case HTTPProblem:
```

Comment:
This would actually be a much more breaking change than initially thought. This would break go-client that passes a non pointer to this function so it effectively breaks all of the service client functions that attempt to pass the non pointer to this function